tests/open_alchemy/helpers/test_peek/test_peek_extension.py

```diff
@@ -646,35 +59,179 @@ def test_mixins(schema, schemas, expected_mixins):
+            ("tablename", True),
+            ("inherits", 1),
+            ("json", 1),
+            ("backref", True),
+            ("secondary", True),
+            ("uselist", "True"),
+            ("foreign-key", True),
+            ("foreign-key-column", True),
+            ("server-default", True),
+            ("dict-ignore", "True"),
+            ("kwargs", True),
+            ("kwargs", {1: True}),
+            ("kwargs", {1: True, "key": "value"}),
+            ("kwargs", {"key": "value", 1: True}),
+            ("kwargs", {1: True, 2: False}),
+            ("foreign-key-kwargs", True),
+            ("foreign-key-kwargs", {1: True}),
+            ("foreign-key-kwargs", {1: True, "key": "value"}),
+            ("foreign-key-kwargs", {"key": "value", 1: True}),
+            ("foreign-key-kwargs", {1: True, 2: False}),
+        ]
+    ],
+)
 @pytest.mark.helper
-def test_dict_ignore_wrong_type():
+def test_peek_malformed_schema_error(prefix, key_values, func):
     """
-    GIVEN schema with dict_ignore defined as a string
-    WHEN dict_ignore is called with the schema
+    GIVEN schema with autoincrement defined as a string
+    WHEN autoincrement is called with the schema
     THEN MalformedSchemaError is raised.
     """
-    schema = {"x-dict-ignore": "True"}
+    schema = {f"{prefix}{key}": value for key, value in key_values}
 
     with pytest.raises(exceptions.MalformedSchemaError):
-        helpers.peek.dict_ignore(schema=schema, schemas={})
+        func(schema=schema, schemas={})
 
 
 @pytest.mark.parametrize(
-    "schema, expected_dict_ignore",
+    "prefix",
+    [pytest.param(p, id=p) for p in helpers.peek.VALID_PREFIXES],
+)
+@pytest.mark.parametrize(
+    "key_values, func",
     [
-        ({}, None),
-        ({"x-dict-ignore": True}, True),
-        ({"x-dict-ignore": False}, False),
+        pytest.param(
+            [(extension, value)],
+            getattr(helpers.peek, extension.replace("-", "_")),
+            id=f"invalid {extension}",
+        )
+        for extension, value in [
+            ("mixins", "MissingDot"),
+            ("mixins", ".second"),
+            ("mixins", "first."),
+            ("mixins", ".second.third"),
+            ("mixins", "first..third"),
+            ("mixins", "first.second."),
+            ("mixins", ["first."]),
+            ("mixins", ["first.", "first.second"]),
+            ("mixins", ["first.second", ".second"]),
+            ("mixins", ["first.", ".second"]),
+            ("composite-index", True),
+            ("composite-unique", True),
+        ]
     ],
-    ids=["missing", "true", "false"],
 )
 @pytest.mark.helper
-def test_dict_ignore(schema, expected_dict_ignore):
+def test_peek_malformed_extension_property_error(prefix, key_values, func):
     """
-    GIVEN schema and expected dict_ignore
-    WHEN dict_ignore is called with the schema
-    THEN the expected dict_ignore is returned.
+    GIVEN schema with autoincrement defined as a string
```

Comment:
should these be generalised (rather than `auto increment` maybe something like `extension property`?

Reply:
Good catch! Addressed ✅